pkg/operators/ebpf/maps.go

```diff
@@ -37,3 +66,216 @@ func (i *ebpfInstance) populateMap(t btf.Type, varName string) error {
 	i.gadgetCtx.SetVar(varName, nilVal)
 	return nil
 }
+
+func (i *ebpfInstance) mapParams() api.Params {
+	if len(i.mapIters) == 0 {
+		return nil
+	}
+	return api.Params{
+		{
+			Key:          ParamMapIterInterval,
+			Description:  "interval in ms in which to iterate over maps and emit",
+			DefaultValue: "1s",
+			TypeHint:     api.TypeString,
+			Title:        "Map fetch interval",
+		},
+		{
+			Key:          ParamMapIterCount,
+			Description:  "number of map fetch cycles - use 0 for unlimited",
+			DefaultValue: "0",
+			TypeHint:     api.TypeInt,
+			Title:        "Map fetch count",
+		},
+	}
+}
+
+func (i *ebpfInstance) evaluateMapParams(paramValues api.ParamValues) error {
+	if len(i.mapIters) == 0 {
+		return nil
+	}
+
+	globalDuration := time.Duration(0)
+	globalCount := 0
+
+	durations, err := apihelpers.GetDurationValuesPerDataSource(paramValues[ParamMapIterInterval])
+	if err != nil {
+		return fmt.Errorf("evaluating map fetch interval: %w", err)
+	}
+	for dsName, duration := range durations {
+		if dsName == "" {
+			globalDuration = duration
+			continue
+		}
+		iter, ok := i.mapIters[dsName]
+		if !ok {
+			return fmt.Errorf("map fetch interval found for non-existing iterator %q", dsName)
+		}
+		iter.interval = duration
+	}
+
+	counts, err := apihelpers.GetIntValuesPerDataSource(paramValues[ParamMapIterCount])
+	if err != nil {
+		return fmt.Errorf("evaluating map fetch interval: %w", err)
+	}
+	for dsName, count := range counts {
+		if dsName == "" {
+			globalCount = count
+			continue
+		}
+		iter, ok := i.mapIters[dsName]
+		if !ok {
+			return fmt.Errorf("map fetch count found for non-existing iterator %q", dsName)
+		}
+		iter.count = count
+	}
+
+	for _, iter := range i.mapIters {
+		if iter.interval == 0 {
+			iter.interval = globalDuration
+		}
+		if iter.count == 0 {
+			iter.count = globalCount
+		}
```

Comment:
I think a good place to add the ds annotations are here:

```suggestion
		}
		iter.ds.AddAnnotation("fetch-count", fmt.Sprintf("%d", iter.count))
		iter.ds.AddAnnotation("fetch-interval", iter.interval.String())
```